sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala

```diff
@@ -88,7 +88,7 @@ class ParquetReadSupport(
       SQLConf.NESTED_SCHEMA_PRUNING_ENABLED.defaultValue.get)
     val parquetFileSchema = context.getFileSchema
     val parquetClippedSchema = ParquetReadSupport.clipParquetSchema(parquetFileSchema,
-      catalystRequestedSchema, caseSensitive)
+      catalystRequestedSchema, caseSensitive, enableVectorizedReader)
```

Comment:
Are all the changes in this file used to provide a better error message for vectorized parquet reader?

Reply:
This is used to throw exceptions instead of returning incorrect result, @cloud-fan .

Reply:
We already throw `SchemaColumnConvertNotSupportedException` as a runtime exception. This is the same approach.